src/test/ui/argument-suggestions/basic.stderr

```diff
@@ -0,0 +1,92 @@
+error[E0308]: arguments to this function are incorrect
+  --> $DIR/basic.rs:20:5
+   |
+LL |     invalid(1.0);
+   |     ^^^^^^^
+   |
+note: expected `u32`, found floating-point number
+  --> $DIR/basic.rs:20:13
+   |
+LL |     invalid(1.0);
+   |             ^^^
+note: function defined here
+  --> $DIR/basic.rs:13:4
+   |
+LL | fn invalid(_i: u32) {}
+   |    ^^^^^^^
```

Comment:
I think that this case is common enough that we might want to special case it slightly. What do you think of

```
error[E0308]: mismatched types: expected `u32`, found floating-point number
  --> $DIR/basic.rs:20:5
   |
LL |     invalid(1.0);
   |     ------- ^^^ expected `u32`, found floating-point number
   |     |
   |     arguments to this function are incorrect
   |
note: function defined here
  --> $DIR/basic.rs:13:4
   |
LL | fn invalid(_i: u32) {}
   |    ^^^^^^^
```

Reply:
The output has changed a bit (the "expected" note is now a label again). I'm on the fence here: I do think we can have a little bit more specific error message when there is only one problem. However, getting this to be "just right" is probably pretty subjective and I don't really want to put in a ton of time on this or have this sit.